transcom/mymove
# B-20948 SSW Calculations INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20948)

## Summary

Final SSW Member-paid and GTCC-paid expenses erroneously including counselor excluded and/or rejected expenses.  If an expense is excluded or rejected by the Closeout Counselor, they should not appear on the Final SSW.

Previously, there was no status check on the moving expenses. Now we check if the expense is in the PPMDocumentStatusRejected or PPMDocumentStatusExcluded status and if it is, we don't include it in the totals for either the Member-paid or GTCC-paid totals on page 2 of the SSW.

### How to test

1. Create a new move with a PPM shipment as the customer (_make sure you request an advance_!)
2. Login as the SC and submit the move.
3. As the customer, upload PPM documents. When you get to the page where you can add expenses, add 2-3
![image](https://github.com/user-attachments/assets/4872c632-9da7-4719-ac1c-0b9c08c72560)
4. Login as the SC and find the move to complete closeout
5. Reject or exclude some (but not all) of the expenses that the customer submitted
![image](https://github.com/user-attachments/assets/48b48033-1021-4216-986f-0c54d34cc67c)

6. After generating the AOA or Payment packet as the office user, verify that you do NOT see the expenses that you rejected/excluded on page 2 of the SSW report.
![image](https://github.com/user-attachments/assets/1d5fa2ff-433d-44e0-accf-f05c1eddeb6b)
7. Check out the screenshots below for a few more test cases.

## Screenshots
Test Case A, no GTCC used, customer submitted two expenses
![image](https://github.com/user-attachments/assets/3b480ec2-d60b-4c55-a85b-6cf21d29bc5b)
SC approved $100 and excluded $50
![image](https://github.com/user-attachments/assets/b23db05d-4413-40ff-8fd3-93a11ef530a8)

Total does not include the excluded/rejected expenses, for a total of $100:
![image](https://github.com/user-attachments/assets/4977b0f8-b69b-4c3b-9dcc-97ff7ce34faf)

Test Case B, GTCC used, customer submitted two expenses
![image](https://github.com/user-attachments/assets/dcdac51f-702e-4c4a-8420-1f6d12b2f5e1)
SC approved $80, rejected $20
![image](https://github.com/user-attachments/assets/ba9cd780-e6fd-4175-ba25-49cc69cf9d4f)
Total on the SSW is $80, rejected $20 not included
![image](https://github.com/user-attachments/assets/f0a1812f-1ac9-4a65-939b-9418d4b2de09)


## Screenshots
Before changes, incorrectly added up the approved as well as excluded/rejected expenses for a total of $150:
![image](https://github.com/user-attachments/assets/3d5632f4-0e16-4f30-bd76-2005e443187d)

After fix, total does not include the excluded/rejected expenses, for a total of $100:
![image](https://github.com/user-attachments/assets/4977b0f8-b69b-4c3b-9dcc-97ff7ce34faf)


No 508 compliance screenshot needed as this only impacts text fields on the generated SSW PDF.
